src/ReadSpeed.cxx

```diff
@@ -0,0 +1,302 @@
+#include "ReadSpeed.hxx"
+
+#include <ROOT/TSeq.hxx>
+#include <ROOT/TThreadExecutor.hxx>
+#include <ROOT/TTreeProcessorMT.hxx>   // for TTreeProcessorMT::GetTasksPerWorkerHint
+#include <ROOT/RDF/InterfaceUtils.hxx> // for ROOT::Internal::RDF::GetTopLevelBranchNames
+#include <TBranch.h>
+#include <TStopwatch.h>
+#include <TTree.h>
+
+#include <algorithm>
+#include <cassert>
+#include <cmath> // std::ceil
+#include <memory>
+#include <stdexcept>
+#include <set>
+#include <regex>
+
+namespace ReadSpeed {
```

Comment:
putting function definitions inside a namespace scope is discouraged, because it's error prone. for example this obviously wrong example compiles fine (until someone actually tries to use `foo`): https://godbolt.org/z/aM5oof8Tj

Instead, function definitions should use a qualified name so the compiler can complain if there is no matching declaration in the header, e.g. `std::vector<std::string> ReadSpeed::GetMatchingBranchNames(...)` instead of just `std::vector<std::string> GetMatchingBranchNames(...)`